src/tracers.jl

```diff
@@ -168,15 +194,18 @@ function additive_merge(a::HessianTracer, b::HessianTracer)
 end
 
 # Merge first- and second-order terms in a "distributive" fashion
-function distributive_merge(a::HessianTracer, b::HessianTracer)
+function distributive_merge(a::HessianTracer{S}, b::HessianTracer{S}) where {S}
     da = deepcopy(a.inputs)
     db = deepcopy(b.inputs)
-    # add second-order interaction term, works by not being clever with symmetry
+    sa = keys2set(S, da)
+    sb = keys2set(S, db)
+
+    # add second-order interaction term by ignoring symmetry
     for (ka, va) in pairs(da)
-        da[ka] = union(va, keys(db))
+        da[ka] = union(va, sb)
     end
     for (kb, vb) in pairs(db)
-        db[kb] = union(vb, keys(da))
+        db[kb] = union(vb, sa)
```

Comment:
These are the most crucial changes to avoid repeated sorting when using `SortedVector`.

Reply:
good one